src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java

```diff
@@ -305,17 +313,106 @@ public void amendOicHearing(final Long adjudicationNumber, final long oicHearing
+        final OicHearingResultRequest oicHearingResultRequest) {
+
+        final var pair = getWithValidationChecks(adjudicationNumber, oicHearingId);
+        final var oicHearing = pair.getLeft();
+        final var adjudication = pair.getRight();
+
+        if (oicHearingResultRepository.findById(new OicHearingResult.PK(oicHearingId, 1L)).isPresent()) {
+            throw new ValidationException(format("Hearing result for hearing id %d already exist for adjudication number %d", oicHearingId, adjudicationNumber));
+        }
+
+        final var staff = staffUserAccountRepository.findByUsername(oicHearingResultRequest.getAdjudicator())
+            .orElseThrow(() -> new EntityNotFoundException(format("Adjudicator not found for username %s", oicHearingResultRequest.getAdjudicator())));
+
+        oicHearing.setAdjudicator(staff.getStaff());
+        oicHearingRepository.save(oicHearing);
+
+        Long oicOffenceId;
+        try {
+            oicOffenceId = adjudication.getOffenderParty().get().getCharges().get(0).getOffenceType().getOffenceId();
+        } catch (Exception e) {
+            throw EntityNotFoundException.withMessage(format("OicOffenceId not found for adjudicationNumber %d and oicHearingId %d", adjudicationNumber, oicHearingId), e.getMessage());
+        }
+
+        final var oicHearingResult = oicHearingResultRepository.save(OicHearingResult.builder()
+            .oicHearingId(oicHearingId)
+            .chargeSeq(1L)
+            .resultSeq(1L)
+            .pleaFindingCode(oicHearingResultRequest.getPleaFindingCode())
+            .findingCode(oicHearingResultRequest.getFindingCode())
+            .agencyIncidentId(adjudication.getAgencyIncidentId())
+            .oicOffenceId(oicOffenceId)
+            .build());
+
+
+        return OicHearingResultDto.builder()
+            .findingCode(oicHearingResult.getFindingCode())
+            .pleaFindingCode(oicHearingResult.getPleaFindingCode())
+            .build();
+    }
+
+    @Transactional
+    @VerifyOffenderAccess
+    public OicHearingResultDto amendOicHearingResult(
+        final Long adjudicationNumber,
+        final Long oicHearingId,
+        final OicHearingResultRequest oicHearingResultRequest) {
+
+        final var oicHearing = getWithValidationChecks(adjudicationNumber, oicHearingId).getLeft();
+
+        final var oicHearingResult = oicHearingResultRepository.findById(new OicHearingResult.PK(oicHearingId, 1L))
+            .orElseThrow(new EntityNotFoundException(format("No hearing result found for hearing id %d and adjudication number %d", oicHearingId, adjudicationNumber)));
+
+        final var staff = staffUserAccountRepository.findByUsername(oicHearingResultRequest.getAdjudicator())
+            .orElseThrow(() -> new EntityNotFoundException(format("Adjudicator not found for username %s", oicHearingResultRequest.getAdjudicator())));
+
+        oicHearing.setAdjudicator(staff.getStaff());
+        oicHearingRepository.save(oicHearing);
```

Comment:
very, very minor but I suspect the `save` are redundant since you are in a transaction where you have done a get from the repo anyway.